File: Recursion/DecimalToBinary/DecimalToBinary.java
import java.util.Scanner;

public class DecimalToBinary {
    private static int decimalToBinary(int num)
    {

        if(num == 0) return 0;
        else return num%2 + decimalToBinary(num/2)*10;
    }

    private static int getOutput() {
        Scanner scanner = new Scanner(System.in);
        int user_inp;
        while(true)
        {
            System.out.println("How to convert a number from Decimal to Binary");

            System.out.print("Enter number: ");
            user_inp = scanner.nextInt();


            if(user_inp>=0) break;
            System.out.println("The Number must be positive integer only");
        }
        return decimalToBinary(user_inp);
    }

    public static void main(String[] args)
    {
        System.out.println(getOutput());
    }
}

File: Recursion/DecimalToBinary/DecimalToBinary.py
# Using Euclidean Algorithm
def decimalToBinary(num):
    assert int(num) == num, 'The parameter must be integer only'
    if num == 0:
        return 0
    else:
        return num%2 + decimalToBinary(num//2)*10



print("How to convert a number from Decimal to Binary")

user_inp = int(input("Enter  number: "))

print(decimalToBinary(user_inp))